jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/events/JettyListenerEventDriver.java

```diff
@@ -160,18 +181,39 @@ public void onTextFrame(ByteBuffer buffer, boolean fin) throws IOException
 
         if (listener instanceof WebSocketPartialListener)
         {
-            if (utf8Partial == null)
+            switch (partialMode)
             {
-                utf8Partial = new Utf8PartialBuilder();
-            }
+                case NONE:
+                    partialMode = PartialMode.TEXT;
+                    // fallthru
+                case TEXT:
```

Comment:
we're trying to validate frame ordering here, but now that `onContinuationFrame()` delegates to `onTextFrame()` we can no longer distinguish between  a `TEXT` frame followed by a `CONTINUATION` frame and two consecutive `TEXT` frames without a fin bit

Reply:
ahh I see that the sequence of frames is also validated in the `Parser`